internal/pkg/util/user/lookup_unix.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// This is the simplified version of Go's os/user/cgo_lookup_unix.go file.
```

Comment:
I would keep the original build tags (even if it's not explicit, I suspect we are building singularity with CGO enabled, otherwise we would have gotten an issue around not working on systems using NIS or LDAP), keep the original names and structure as much as possible (with the original errors, too), _not_ do the shell patching here and modify `buildUser` so that it uses `Gecos` instead of `Name` and it doesn't edit the Gecos information to extract the name.

Add a _different_ `.go` file that calls `buildUser` or `lookupUser` as necessary and do the shell patching there (look at the original `convertUser`, I get the impression we lost that in this change, if it's still there I'm not finding it).

Reply:
After some thoughts I still think we need to keep shell patching and try to get rid of  unnecessary int/string casting. 
However, I agree that it is reasonable to keep build tags and file structure alike.